integration_tests/default-config_test.go

```diff
@@ -156,4 +160,64 @@ var _ = When("create default backstage", func() {
 
 	})
 
+	It("creates runtime object using raw configuration then updates StatefulSet to replace some immutable fields", func() {
+		if !*testEnv.UseExistingCluster {
+			Skip("Real cluster required to assert actual deletion and replacement of resources")
+		}
+
+		rawStatefulSetYamlContent := readTestYamlFile("raw-statefulset.yaml")
+		dbConf := map[string]string{"db-statefulset.yaml": rawStatefulSetYamlContent}
+
+		dbRaw := generateConfigMap(ctx, k8sClient, "dbraw", ns, dbConf, nil, nil)
+
+		backstageName := createAndReconcileBackstage(ctx, ns, bsv1alpha1.BackstageSpec{
+			RawRuntimeConfig: &bsv1alpha1.RuntimeConfig{
+				LocalDbConfigName: dbRaw,
+			},
+		}, "")
+
+		Eventually(func(g Gomega) {
+			By("creating Deployment")
+			deploy := &appsv1.Deployment{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: model.DeploymentName(backstageName)}, deploy)
+			g.Expect(err).ShouldNot(HaveOccurred())
+
+			By("creating StatefulSet")
+			ss := &appsv1.StatefulSet{}
+			name := fmt.Sprintf("backstage-psql-%s", backstageName)
+			err = k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: name}, ss)
+			g.Expect(err).ShouldNot(HaveOccurred())
+			g.Expect(ss.Spec.Template.Spec.Containers).To(HaveLen(1))
+			g.Expect(ss.Spec.Template.Spec.Containers[0].Image).To(Equal("busybox"))
+			g.Expect(ss.Spec.ServiceName).To(Equal("backstage-psql-db-statefulset-hl"))
+		}, time.Minute, time.Second).Should(Succeed())
+
+		By("updating CR with new DB raw config")
```

Comment:
Up to you, but, why do you need to update raw config?
IMO it is enough to just update Backstage with default config (especially after https://github.com/janus-idp/operator/pull/371 ) no?

Reply:
I wanted to check that something had actually changed to ensure the StatefulSet replacement. So it should be a field that is not overwritten upon the first reconciliation.
But with #371 in mind, I'll change that to another immutable field and update Backstage to the default config. Thanks.

Reply:
Updated in [`0f194b2` (#369)](https://github.com/janus-idp/operator/pull/369/commits/0f194b28ebcb5f9374d96c8e3ba781484c23fd5b). PTAL.